test/unittests/mongoBackend/mongoContextProvidersUpdate_test.cpp

```diff
@@ -84,6 +88,34 @@
 
 /* ****************************************************************************
 *
+* getAttr -
+*
+* We need this function because we can not trust on array index, at mongo will
+* not sort the elements within the array. This function assumes that always will
+* find a result, that is ok for testing code.
+*
+* FIXME P7: duplicated function in mongoUpdateContext_test.cpp. Unify.
+*/
+static BSONObj getAttr(std::vector<BSONElement> attrs, std::string name, std::string type, std::string id = "") {
```

Comment:
This is just a unit-test function, but try to avoid pushing big objects on the stack.
Especially vectors of big objects ...

    static BSONObj getAttr(const std::vector<BSONElement>& attrs, const std::string& name, const std::string& type, const std::string& id = "")